src/components/career-form/career-file-upload.tsx

```diff
@@ -0,0 +1,238 @@
+      return {
+        code: 'file is too big',
+        message: `Maximal filesize of 20 MB is reached.`,
+      };
+    }
+    return null;
+  };
+
+  const onDropRejected = (file) => {
+    setError(
+      name,
+      { type: 'manual', message: file[0].errors[0].message },
+      { shouldFocus: true },
+    );
+  };
+
+  const { getRootProps, getInputProps, isDragActive } = useDropzone({
+    onDrop,
+    accept: '.pdf',
+    validator,
+    onDropRejected,
+  });
+
+  const unselectFile = (event, index) => {
+    const dataTransfer = new DataTransfer();
+    for (let i = 0; i < selectedFiles.length; i++) {
+      if (i != index) {
+        dataTransfer.items.add(selectedFiles[i]);
+      }
+    }
+    setValue('documents', dataTransfer.files, { shouldDirty: true });
+    clearErrors(name);
+    event.stopPropagation();
+    event.preventDefault();
+  };
+
+  return (
+    <>
+      {/*File-Upload*/}
+      <GridItem>
+        <InputContainer>
+          <FileInputLabel
+            {...getRootProps({
+              selectedFiles,
+              isDragActive,
+              onClick: (event) => event.stopPropagation(),
+            })}
+            hasError={errors.documents}
+          >
+            <FileInput
+              id={name}
+              type={'file'}
+              {...getInputProps({
+                ...register(name, { required: 'Dein CV fehlt' }),
+              })}
+              multiple
+            />
+            {children}
+          </FileInputLabel>
+        </InputContainer>
+        <FormError error={errors.documents} marginBottom={8} />
+      </GridItem>
+
+      {/*File-Review*/}
+      <GridItem>
+        {selectedFiles &&
+          selectedFiles.length > 0 &&
+          Object.entries(selectedFiles).map(([index, file]) => {
+            if (index !== 'length') {
+              return (
+                <GridItem key={index}>
+                  <FilePreview index={index} onClick={unselectFile}>
+                    <FileTitleWrapper file={file as File} />
+                    <CareerSelect
+                      errors={errors}
+                      file={file}
+                      register={register}
+                      setValue={setValue}
+                      clearError={clearErrors}
+                      watch={watch}
+                    />
+                  </FilePreview>
+                </GridItem>
+              );
+            }
+          })}
+      </GridItem>
+    </>
+  );
+};
+
+const FileInputLabel = styled.label<FileUploadProps & DropzoneRootProps>`
+  display: flex;
+  flex-direction: column;
+  justify-content: center;
+  ${({ selectedFiles }) =>
+    (!selectedFiles || selectedFiles.length === 0) && `align-items: center;`}
+  border-radius: 4px;
+  ${({ isDragActive }) =>
+    isDragActive && `border: 2px solid ${theme.palette.primary.main};`}
+  padding: ${({ selectedFiles }) =>
+    !selectedFiles || selectedFiles.length === 0 ? `56px` : `24px`};
+  margin-bottom: 2px;
```

Comment:
I love styled component but that is just ugly and barely readable  😑 I see myself writing code like that and maybe it would help to pre-caculate `!selectedFiles || selectedFiles.length === 0)` into a boolean `empty` and pass that instead as a prop?